### DISCUSSION TITLE
New File does not send workspace/didCreateFiles notification

### DISCUSSION THREAD

**Discussion by aghayanlena** [eclipse-theia/theia#13687] [General]
I have my custom Theia application integrated with my custom LSP Server vs code extension. 

Theia does not send a workspace/didCreateFiles notification from the New File in the 'File' menu bar. The notification is sent only when the new file is created from the workspace toolbar. 

I would like to check if this is an expected behavior. I don't see the same behavior in VS code.

**Comment by msujew** [comment]
Hey @aghayanlena,

looking a this, I've noticed that we don't fire the `onDidCreateFiles` event in the plugin host when running a `saveAs` operation, which creates a new file. This is the operation that is used when creating a new file and saving it to disk. I've created https://github.com/eclipse-theia/theia/issues/13688 to track this bug.

**Reply by msujew** [reply]
@aghayanlena FYI, I've created PR https://github.com/eclipse-theia/theia/pull/13689 to fix this.

**Comment by sophieminasyan** [comment]
@msujew 

I've updated Theia to version 1.52.0, but I'm still experiencing an issue. Theia does not send a ՛_workspace/didCreateFiles_՛ notification when creating a new file from the 'File' menu.
 
Could you please verify if this issue still exists?
